ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/container.html

```diff
@@ -14,6 +14,11 @@
   ~ limitations under the License.
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
 <sly data-sly-test="${wcmmode.edit}" data-sly-use.allowed="com.day.cq.wcm.foundation.AllowedComponentList"></sly>
+<div data-sly-test="${wcmmode.disabled || wcmmode.preview}" data-sly-unwrap>
+     <script>
+         document.cookie="wcmmode=disabled; max-age=0; path=/";  
```

Comment:
I recall Sites moved away from Cookie couple of years back - not sure if they have moved back - but agree on avoiding cookie..

Reply:
@iamsudhanshu, @rismehta  Core Editor is setting wcmmode cookie, and there is WCMMode filter which provides WCM mode for each servlet request in the following order

query param
request attribute
cookie
Actually, we are just removing cookie set by core editor here 😃 , see **max-age 0**
cc: @sakshi-arora1

Reply:
@pankaj-parashar In general, it is not recommended to add scripts in core component. All JS is part of footer for a better LHS score.

Reply:
Can we rather set this cookie from the server via Set-Cookie ?

Reply:
This code is an open space, these are internal details which should not be exposed to customer. Ideally, such things needs to be handled on the server

Reply:
I thought of handling it server side, but in case cached form page html is served, the solution will not work. In case this page HTML is served from CDN/Dispatcher, server-side logic of set/remove cookie will not come into the picture.

Reply:
can we add this in runtime clientlib, that is not loaded in authouring and is available in both preview and disabled.
cc: @pankaj-parashar @rismehta

Reply:
>  but this is something related to guide container loading, not to the page or header/footer

There are many things related to guideContainer (like custom client libraries) which we add only on footer via `FormStructureParser` for better core web vitals.

Reply:
why did I add this script in the container rendition?
I am not aware of form container internals, and in case any subsequent/intermediate calls are made while the guide container is loading or before the container is loaded, wcmmode cookie should be deleted, so all these calls don't have incorrect wcmmode. This was the purpose for including the script here in the form container. If we are sure that all subsequent calls (except css/js) will be made after form container/page loading, I can move this logic to runtime clientlib as suggested by @sakshi-arora1 
cc: @rismehta, @iamsudhanshu